### ISSUE TITLE
BUG: Script's default "__main__" Test conflicts with Template's "__main__" test.

### ISSUE THREAD

**Issue by donovankeith** [NiklasRosenstein/c4d-prototype-converter#27]
Console Error:

```
 Traceback (most recent call last):
  File "'paintfracture.pyp'", line 153, in <module>
NameError: name 'main' is not defined
```

Test in C4D's default script:

```
if __name__=='__main__':
  main()
```

Test in `c4d-prototype-converter`'s script template:

```
if __name__ == '__main__':
  PaintFracture().Register()
```

**Comment by NiklasRosenstein** [comment]
I'm not quite sure what you are trying to say.

Currently `if __name__ == '__main__': main()` portion from a script should be removed manually after the conversion. I may be able to remove it automatically, not sure how reliable that will be though.

Converting the C4D default script should *currently* result in this:

```python
# Copyright (c) <year> <author>

import c4d
import os


def load_bitmap(path):
  path = os.path.join(os.path.dirname(__file__), path)
  bmp = c4d.bitmaps.BaseBitmap()
  if bmp.InitWith(path)[0] != c4d.IMAGERESULT_OK:
    bmp = None
  return bmp


import c4d
from c4d import gui
if __name__=='__main__':
  main()


class untitled(c4d.plugins.CommandData):

  PLUGIN_ID = 43534
  PLUGIN_NAME = 'untitled'
  PLUGIN_INFO = 0
  PLUGIN_ICON = load_bitmap('res/icons/untitled.tiff')
  PLUGIN_HELP = ""

  def Register(self):
    return c4d.plugins.RegisterCommandPlugin(
      self.PLUGIN_ID, self.PLUGIN_NAME, self.PLUGIN_INFO, self.PLUGIN_ICON,
      self.PLUGIN_HELP, self)

  #Welcome to the world of Python
  
  
  def Execute(self, doc):
    gui.MessageDialog('Hello World!')
  
    return True


if __name__ == '__main__':
  untitled().Register()
```

**Comment by donovankeith** [comment]
Gotcha. I think I was just expecting things to run immediately after conversion. I do think that if the code block is exactly:

```
if __name__=='__main__':
  main()
```

With nothing following `main()` in that block, then it should just be omitted.

Same for `import c4d`. This way for simple scripts, you get a running plugin with no additional work.

**Comment by donovankeith** [comment]
Or, if you don't feel comfortable with that, add something like:

```
# TODO: Delete this after copying any essential code from this conditional to the same conditional found at the bottom of this file.
if __name__=='__main__':
  main()
```

**Comment by NiklasRosenstein** [comment]
Resolved with #29
